107/src/main/java/org/ehcache/jsr107/Eh107CacheManager.java

```diff
@@ -157,36 +151,35 @@ public Properties getProperties() {
         return cache;
       }
 
-      // copy the config (since it can be re-used per 107 spec)
-      Eh107CompleteConfiguration<K, V> completeConfig = new Eh107CompleteConfiguration<K, V>(config);
-
-      CacheResources<K, V> cacheResources = new CacheResources<K, V>(cacheName, completeConfig);
-      Eh107Expiry<K, V> expiry = cacheResources.getExpiryPolicy();
+      ConfigurationMerger.ConfigHolder<K, V> configHolder = configurationMerger.mergeConfigurations(cacheName, config);
 
       final org.ehcache.Cache<K, V> ehCache;
       try {
-        ehCache = ehCacheManager.createCache(cacheName,
-            toEhcacheConfig(cacheName, completeConfig, cacheResources, expiry));
+        ehCache = ehCacheManager.createCache(cacheName, configHolder.cacheConfiguration);
       } catch (Throwable t) {
         // something went wrong in ehcache land, make sure to clean up our stuff
         // NOTE: one relatively simple error path is if a pre-configured cache of the same name already exists in
         // ehcache
         MultiCacheException mce = new MultiCacheException(t);
-        cacheResources.closeResources(mce);
+        configHolder.cacheResources.closeResources(mce);
         throw mce;
       }
 
       Eh107Cache<K, V> cache = null;
       try {
-        cache = new Eh107Cache<K, V>(cacheName, new Eh107CompleteConfiguration<K, V>(config, ehCache.getRuntimeConfiguration()), cacheResources, ehCache, this, expiry);
+        if (configHolder.useEhcacheLoaderWriter) {
+          configHolder.cacheResources = new CacheResources<K, V>(cacheName, EhcacheHackAccessor.getCacheLoaderWriter((Ehcache<K, V>)ehCache), configHolder.cacheResources.getExpiryPolicy(), configHolder.cacheResources.getListenerResources());
```

Comment:
Current logic says so, will do it or refactor logic so that it is not a concern (not initialised)